### PR TITLE
contractParmasConvertToBytes

### PR DESCRIPTION
**Pull request by himanshuoodles**
I have added a function to get bytecode of parmas of contract . So at the time of sending RawTransaction the user can get bytecode + parmsBytes as data to the RawTransaction

### CODE REVIEW

File: lib/web3/contract.js [hunk 1]
```diff
@@ -176,6 +176,35 @@ var ContractFactory = function (eth, abi) {
      * @param {Function} callback
      * @returns {Contract} returns contract instance
      */
+    
+    
+      //   this.customByteCodeConvert
+
+  this.contractParmasConvertToBytes = function () {
```
Comment (himanshuoodles) [inline lib/web3/contract.js]: Yes this whole code is copy and paste , but there should be any function which gives bytecode for parameter of the contract new function . So i have introduce it.

**Comment by frozeman** [comment]
This will all be present in 1.0 in a different way. im not sure if adding new functionality in 0.18.2 makes sense.
Please see the new docs of the coming version: http://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#methods-mymethod-encodeabi

File: lib/web3/contract.js [hunk 2]
```diff
@@ -176,6 +176,35 @@ var ContractFactory = function (eth, abi) {
...
+      }
+
+      var last = args[args.length - 1];
+      if (utils.isObject(last) && !utils.isArray(last)) {
+          options = args.pop();
+      }
+    //  console.log("inside web3 args "+args,this.abi);
+      var bytes = encodeConstructorParams(this.abi, args);
+    //  options.data += bytes;
+    return callback(null,bytes);
```
Comment (beether) [inline lib/web3/contract.js]: Why does this function take a callback?  It can just return bytes.

**Comment by frozeman** [comment]
Thanks for the effort  I'm closing this in favor of 1.0

File: lib/web3/contract.js [hunk 3]
```diff
@@ -176,6 +176,35 @@ var ContractFactory = function (eth, abi) {
...
+
+  this.contractParmasConvertToBytes = function () {
+      var contract = new Contract(this.eth, this.abi);
+
+      // parse arguments
+      var options = {}; // required!
+      var callback;
+
+      var args = Array.prototype.slice.call(arguments);
+      //  console.log("inside web3 args "+args);
```
Comment (wizzfile) [inline lib/web3/contract.js]: Why

File: lib/web3/contract.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (wizzfile) [reply lib/web3/contract.js]: > +      var callback;
+
+      var args = Array.prototype.slice.call(arguments);
+      //  console.log("inside web3 args "+args);
+      if (utils.isFunction(args[args.length - 1])) {
+          callback = args.pop();
+      }
+
+      var last = args[args.length - 1];
+      if (utils.isObject(last) && !utils.isArray(last)) {
+          options = args.pop();
+      }
+    //  console.log("inside web3 args "+args,this.abi);
+      var bytes = encodeConstructorParams(this.abi, args);
+    //  options.data += bytes;
+    return callback(null,bytes);

File: lib/web3/contract.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (wizzfile) [reply lib/web3/contract.js]: > +      var callback;
+
+      var args = Array.prototype.slice.call(arguments);
+      //  console.log("inside web3 args "+args);
+      if (utils.isFunction(args[args.length - 1])) {
+          callback = args.pop();
+      }
+
+      var last = args[args.length - 1];
+      if (utils.isObject(last) && !utils.isArray(last)) {
+          options = args.pop();
+      }
+    //  console.log("inside web3 args "+args,this.abi);
+      var bytes = encodeConstructorParams(this.abi, args);
+    //  options.data += bytes;
+    return callback(null,bytes);

**Pull request closed without merging**